lisa/sut_orchestrator/azure/common.py

```diff
@@ -2225,9 +2236,12 @@ def get_deployable_vhd_path(
     """
     matches = SAS_URL_PATTERN.match(vhd_path)
     if not matches:
-        vhd_details = get_vhd_details(platform, vhd_path)
+        vhd_details = get_vhd_details(platform, vhd_path, vhd_subscription)
         vhd_location = vhd_details["location"]
-        if location == vhd_location:
+        if (
+            location == vhd_location
+            and vhd_details["subscription"] == platform.subscription_id
```

Comment:
vhd_details may not contain subscription

Reply:
Have added.